### PR TITLE
feat: use security-group module instead of resource

### PR DESCRIPTION
**Pull request by SweetOps**
## what
* use security-group module instead of resource
* update tests

## why
* more flexible  than current implementation 
* bring configuration of security group/rules to one standard

## references
* CPCO-409

### CODE REVIEW

**Review by korenyoni — DISMISSED** [review]
Looks good to me, especially considering https://github.com/cloudposse/terraform-aws-elastic-beanstalk-environment/pull/166 and https://github.com/cloudposse/terraform-aws-elasticache-redis/pull/119

File: variables.tf [hunk 1]
```diff
@@ -8,10 +8,47 @@ variable "subnet_ids" {
...
+  ]
+  description = <<-EOT
+    A list of maps of Security Group rules. 
+    The values of map is fully complated with `aws_security_group_rule` resource. 
+    To get more info see https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group_rule .
+  EOT
+}
+
+variable "security_groups" {
+  description = "A list of Security Group IDs to associate with ALB."
```
Comment (korenyoni) [inline variables.tf:49]:
```suggestion
  description = "A list of Security Group IDs to associate with the ALB."
```

File: variables.tf [hunk 2]
```diff
@@ -8,10 +8,47 @@ variable "subnet_ids" {
   description = "A list of subnet IDs to associate with ALB"
 }
 
-variable "security_group_ids" {
+variable "security_group_enabled" {
+  type        = bool
+  description = "Whether to create default Security Group for ALB."
```
Comment (korenyoni) [inline variables.tf:13]:
```suggestion
  description = "Whether to create default Security Group for the ALB."
```

**Review by korenyoni — COMMENTED** [review]
Not worth me undoing the approval, but consider adding `the` where specified to the variable descriptions, either in this PR or in the future.

**Review by korenyoni — APPROVED** [review]

**Comment by nitrocode** [comment]
Please also add release notes similar to https://github.com/cloudposse/terraform-aws-efs/releases/tag/0.31.0 prior to merging.

**Comment by nitrocode** [comment]
Closing for now as this will need refactoring when new sg changes take affect.

**Pull request closed without merging**